### ISSUE TITLE
Tag `source:opening_hours` should be deleted when updating a new place in the Places overlay

### ISSUE THREAD

**Issue by jdejaegh** [streetcomplete/StreetComplete#6057]
When a shop changed, the Places overlay does not delete the `source:opening_hours` tag from the object.  
The `source:opening_hours` is a documented tag: https://wiki.openstreetmap.org/wiki/Key:source:opening_hours

**How to Reproduce**
1. Find a shop that has at least those tags: `name`, `opening_hours`, `source:opening_hours`
2. Using the Places overlay, change the name of the shop and confirm that it is a new place
3. Observe that `name` is changed, `opening_hours` is deleted and `source:opening_hours` remains

**Expected Behavior**
The tag `source:opening_hours` should be deleted.

**Versions affected**
StreetComplete v60.0
Android 13

**Comment by matkoniecz** [comment]
Can you link an effected object?

**Comment by matkoniecz** [comment]
https://github.com/mnalis/StreetComplete-taginfo-categorize would need to be updated to fix it, if it happens

**Comment by jdejaegh** [comment]
> Can you link an effected object?

Sure, here is one: https://www.openstreetmap.org/node/1210562035/history/13

Version 13 was created by StreetComplete (without deleting the tag `source:opening_hours`).  I manually deleted the tag in version 14.

**Comment by westnordost** [comment]
Hmm, after looking at the PR, I am not sure if removing `source:<key>` when `<key>` is deleted is the wisest thing to do automatically. Because:

1. it opens a can of worms. There are quite a few more keys that are considered _meta keys_. Some more so, some less, some rather only for a certain group of features, etc.. If `source:` is treated specially, for consistency, so should others (`note:`?), but we will have the recurring discussion whether that _meta key_ can always be safely deleted when the key is deleted.

2. Even for `source` which is somewhat seen as a meta key, there are outliers, most prominently, `source:maxspeed`, which can hold valuable data also without an accompanying `maxspeed` value. (`source:maxspeed` is used 2.4M times, `source:opening_hours` is used 4k times) `source:maxspeed` should not be deleted when `maxspeed` is deleted.

So, in summary, I think it is too risky to treat meta keys in a special way.

I think it is best to treat such keys like any other. @mnalis, which `source:`... keys are generally in (enough) use to be included in _StreetComplete-taginfo-categorize_?

**Comment by mnalis** [comment]
Yes, now that you mention that @westnordost , exceptions like `source:maxspeed` would be problematic to be removed when `maxspeed` is removed. So [that suggestion](https://github.com/streetcomplete/StreetComplete/pull/6067#issuecomment-2566397094) of always removing those meta-tags when ***any*** tag is removed is indeed potentially problematic, so I've aborted work on that and rollbacked.

---

**However @westnordost , how about the previous (original) form of the PR, which just removed `source:xxx` and `check_date:xxx` if `xxx` was being removed due to shop being removed/replaced, and `xxx` being listed in `KEYS_THAT_SHOULD_BE_REMOVED_WHEN_PLACE_IS_REPLACED` ?**

E.g. when `amenity=cafe` is being removed, we currently also remove `smoking=*`, and that PR would also remove `source:smoking=*` and `check_date:smoking=*` (there are few hundred instances of that example combined worldwide -- many others are in dozens uses, and there are thousands of `source:*` tags, so that should work IMHO). 
I've glanced over all the tags in `KEYS_THAT_SHOULD_BE_REMOVED_WHEN_PLACE_IS_REPLACED` and I didn't notice anything that should be problematic if its `source:*` or `check_date:*` was removed too. (e.g. no `source:maxspeed`-alike outliers), so I think it should be fine. Can't foresee potential unforeseeable issues, of course :smiley_cat:  

I would keep it to those two meta keys, `note:`  might perhaps be ambiguous (e.g. `note:smoking` on polygon which is both `building` and `shop`, and we're removing the shop).

---

> which source:... keys are generally in (enough) use to be included in StreetComplete-taginfo-categorize?

If you dislike that idea above of removing `check_date:` and `source:` for `KEYS_THAT_SHOULD_BE_REMOVED_WHEN_PLACE_IS_REPLACED`, manual categorization is possibility too. I'd have to update the script to scan those too, and then see how many are there to categorize (there are at least several thousands of `source:*` and `check_date:*` tags; they'd would then need to be assessed for their usages). 
But note that the list might double or triple in size then.
